docs/weave/weave-api-spec/02-transaction-signatures.md

```diff
@@ -0,0 +1,60 @@
+---
+id: tx-sign-spec
+title: Transaction signitures 
+sidebar_label: Transaction signatures 
+---
+
+In this section, we will explain how to sign __transactions__.
+
+## Authentication and Authorization
+
+To execute desired actions and state changes on weave based blockchains, every transaction must be validated, authenticated and then authorized to perform the action. In the realm of blockchains this is achieved by using signatures.
+
+Weave uses `ed25519` signing algorithm to achieve this important feature. 
+After following the steps in [Weave transactions](transaction.md), created **Tx** must be signed. But first Weave *conditions* role in signing must be explained.
+
+[//]: # (TODO move readthedocs documentation to this project and give references here)
+
+Conditions format defined as `(extension, type, data)` in general. 
+
+Condition with __ed25519__ public key signature could be represented as `("sigs", "ed25519", <address>)`. 
+
+Here is the basic golang presentation of condition 
+
+`condition := sprintf("%s/%s/%X", extension, type, data)` where __condition__ looks something like `cond:sigs/ed25519/636f6e646974696f6e64617461`.
+
+## Signing transactions
+
+After the steps in [Weave transactions](transaction.md) are followed, created **Tx** is ready to be signed to be authenticated and authorized by *Weave/bnsd*. 
+
+Signature input body format is:
+
+| version     | len(chainID)     | chainID          | nonce                 | signBytes                  |
+|--------:    |--------------    |--------------    |-------------------    |------------------------    |
+| 4bytes      | uint8            | ascii string     | int64 (bigendian)     | serialized transaction     |
+
+- **version** is the current way to prefix the bytes. Its current value is `00000000 11001010 11111110 00000000`. As hex `0, 0xCA, 0xFE, 0`. Please refer to [iov-one/weave/sigs/SignCodeV1](https://github.com/iov-one/weave/blob/v0.18.0/x/sigs/controller.go#L14).
+
+After this arbitrary bytes put together, this is then prehashed with __sha512__ before fed into
+the public key signing/verification step. This step is applied in order to guarantee signing could be done by a ledger. 
+
+After having the hashed transaction and signed with the wallet's **ed25519** private key, you have your **signed transaction** at your will.
+
+Weave signing algorithm basic representation:
```

Comment:
Previous code examples were written in Go, so I expected that this one is also in Go. Maybe it is worth adding what this example is written in (pseudocode?)

Reply:
I intentionally wrote it as pseudocode so non go and js developers can read and understand it without any language specific questions wont pop up